File: 1720-crawler-log-folder/crawler-log-folder.dart
enum Log {
  child,
  parent,
  current,
  ;

  factory Log.fromString(String string) {
    switch (string) {
      case '../':
        return Log.parent;
      case './':
        return Log.current;
      default:
        return Log.child;
    }
  }
}

class Solution {
  int minOperations(List<String> logs) {
    var currentDepth = 0;
    for (final log in logs) {
      switch (Log.fromString(log)) {
        case Log.child:
          currentDepth++;
        case Log.parent:
          currentDepth > 0 ? currentDepth-- : currentDepth += 0;
        case Log.current:
          break;
      }
    }
    return currentDepth;
  }
}

File: 1720-crawler-log-folder/crawler-log-folder.go
func minOperations(logs []string) int {
    currentDepth := 0

    for _, log := range logs {
        switch log {
        case "../":
            if currentDepth > 0 {
                currentDepth--
            }
        case "./":
        default:
            currentDepth++
        }
    }

    return currentDepth
}
